### PR TITLE
Fixing automation interpolation/lackthereof for bokeh/pyqtgraph

### PR DESCRIPTION
**Pull request by jibarnum**
Fixing issues brought up in ticket #72

### CODE REVIEW

File: pytplot/QtPlotter/TVarFigure1D.py [hunk 1]
```diff
@@ -105,11 +106,95 @@ def _visdata(self):
...
+                    # Flatten the overall_list to just one list, instead of a list of many lists
+                    overall_list = [y for x in overall_list for y in x]
+
+                    # Remove data gaps removed based on user-input acceptable time gap
+                    # In order to do this, we set the identified indices from overall_list to 0, which in the
+                    # connect keyword argument in self.plotwindow.plot will cause that point to not be plotted
+                    time_filtered = np.array([1]*len(dataset.data.index.tolist()))
+                    for index, pos in enumerate(overall_list):
+                        # Remove unwanted time from being plotted
+                        time_filtered[pos] = 0
```
Comment (bryan-harter) [inline pytplot/QtPlotter/TVarFigure1D.py]: If this is doing what I think it is doing, you can set the indexes in time_filtered without looping by using the expression

time_filtered[overall_list] = 0

File: pytplot/QtPlotter/TVarFigure1D.py [hunk 2]
```diff
@@ -105,11 +106,95 @@ def _visdata(self):
...
+                    index = 0  # What index in the below for loop we're on
+                    nans = dict()  # Where nan value indices + times associated with those indices are housed
+                    for j in dataset.data[i].keys():
+                        point = dataset.data[i][j]  # Just grab the point in that dataset, at a specific time
+                        if np.isnan(point):
+                            nans[index] = j  # If we have a nan value, put the index/time in the nan dictionary
+                        index += 1
+                    # Putting the keys & values from the nans dictionary to lists to make them more easily accesible
+                    nan_keys = list(nans.keys())
+                    nan_values = list(nans.values())
```
Comment (bryan-harter) [inline pytplot/QtPlotter/TVarFigure1D.py]: You could replace the previous ~10 lines with:

dataset.data[i][dataset.data[i].isnull().values].index

Not sure if it is faster or not, any I haven't looked into how you would get the keys yet

Comment (jibarnum) [reply pytplot/QtPlotter/TVarFigure1D.py]: Not opposed to getting rid of this loop at all. haha I'll think about how to combine that idea and the keys/values to make it happen....

File: pytplot/QtPlotter/TVarFigure1D.py (hunk 1 above)
Comment (jibarnum) [reply pytplot/QtPlotter/TVarFigure1D.py]: For some reason I feel like I had done just that before, and then switched for some reason... although it likely was a result of something else and i just kept the loop even though it isn't needed.

File: pytplot/QtPlotter/TVarFigure1D.py (hunk 2 above)
Comment (jibarnum) [reply pytplot/QtPlotter/TVarFigure1D.py]: Turns out another list comprehension is the answer for keys.

**Comment by bryan-harter** [comment]
Merging!

**Pull request merged**